src/runtime/agas/addressing_service.cpp

```diff
@@ -2352,6 +2353,8 @@ void addressing_service::start_shutdown(error_code& ec)
     mutex_type::scoped_lock l(refcnt_requests_mtx_);
     enable_refcnt_caching_ = false;
     send_refcnt_requests_sync(l, ec);
+
+    shutdown_started_ = true;
```

Comment:
The runtime has a state variable which can be used to identify whether the shutdown has started. Should we use that instead? However I guess this is absolutely minor.